logstore/lstoreds/keybook.go

```diff
@@ -175,18 +175,17 @@ func (kb *dsKeyBook) ClearKeys(t thread.ID) error {
 
 // ClearLogKeys deletes all keys under a log.
 func (kb *dsKeyBook) ClearLogKeys(t thread.ID, p peer.ID) error {
-	if err := kb.clearKeys(dsLogKey(t, p, kbBase).Child(privSuffix)); err != nil {
```

Comment:
Relied on the assumption that prefix will contain the exact key match.
Switched to simply deleting directly.